src/Core/src/Hosting/MauiAppBuilder.cs

```diff
@@ -104,6 +109,12 @@ ILoggingBuilder InitializeLogging()
 			}
 		}
 
+		IDictionary<object, object> IHostApplicationBuilder.Properties => throw new NotImplementedException();
+
+		IHostEnvironment IHostApplicationBuilder.Environment => throw new NotImplementedException();
```

Comment:
This is often loaded from an environment variable or some json configuration. We could (somehow) detect this as either Debug or Release, or set something during `dotnet publish`, but it is a new concept.

> Gets information about the hosting environment an application is running in.

Reply:
Yeah, this is (2) that I called out in https://github.com/dotnet/maui/issues/4393. I'm not sure what to do here either. Maybe just always returning `Production`? Or `Development` when the app is build for `Debug` and `Production` otherwise?